gtsfm/evaluation/metrics_report.py

```diff
@@ -0,0 +1,205 @@
+                        q3=[quartiles["q3"]],
+                        lowerfence=[quartiles["q0"]],
+                        upperfence=[quartiles["q4"]],
+                        name=metric_name,
+                    ),
+                    row=row,
+                    col=col,
+                )
+
+    # Return the figure converted to HTML.
+    return fig.to_html(full_html=False, include_plotlyjs="cdn")
+
+
+def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the mean,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module.
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    scalar_metrics = {}
+    metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
+
+    # Separate the scalar metrics.
+    for metric, value in metrics_dict.items():
+        if isinstance(value, dict):
+            # Metrics with a dict representation must contain a summary.
+            if not metrics.SUMMARY_KEY in value:
+                raise ValueError("Metric {metric} does not contain a summary.")
+            # Add a scalar metric for mean of 1D distributions.
+            scalar_metrics["mean_" + metric] = value[metrics.SUMMARY_KEY]["mean"]
+        else:
+            scalar_metrics[metric] = value
+    table = create_table_for_scalar_metrics(scalar_metrics)
+    plots_fig = create_plots_for_distributions(metrics_dict)
+    return table, plots_fig
+
+
+def get_html_metric_heading(metric_name: str) -> str:
+    """Helper to get the HTML heading for a metric name.
+
+    This converts a "metric_name" to "Metric Name" (for readability) and adds some style.
+
+    Returns:
+        HTML for heading as a string.
+    """
+    metric_name = get_readable_metric_name(metric_name)
+    metric_html = f'<p style="font-size:25px;font-family:Arial">{metric_name}</p>'
+    return metric_html
+
+
+def get_html_header() -> str:
+    """Helper to get a HTML header with some CSS styles for tables.""
+
+    Returns:
+        A HTML header as string.
+    """
+    return """<head>
+                <style>
+                  table {
+                    font-family: arial, sans-serif;
+                    border-collapse: collapse;
+                    width: 768px
+                  }
+                  td, th {
+                    border: 1px solid #999999;
+                    text-align: left;
+                    padding: 8px;
+                  }
+                  tr:nth-child(even) {
+                    background-color: #dddddd;
+                  }
+                </style>
+              </head>"""
+
+
+def save_metrics_report_html(metrics_groups: List[GtsfmMetricsGroup], html_path: str) -> None:
```

Comment:
maybe name it as `report_metrics_as_html`?

Reply:
i prefer `save_metrics_as_html_report()`

Reply:
Yeah, I see what you mean Ayush, but I think `report` is quite abstract. I think `generate_metrics_report_html` would be a better name, since its actually generating the HTML and not just saving it.